samples/drivers/ublox_sara_n310/README.rst

```diff
@@ -0,0 +1,49 @@
+.. _ublox_sara_n310:
+
+SARA-N310 modem driver sample
+########################################################
+
+Overview
+********
+
+This is an application that tests basic socket functionality of the N310 modem. The following network functions are tested in the application loop:
+- socket()
```

Comment:
Within the modem driver you essentially tell the OS that the modem driver will take care of socket offloading (like NET_DEVICE_DT_INST_OFFLOAD_DEFINE() and NET_SOCKET_REGISTER()). This is done upon initialization of the driver (so on boot) and uses the net sockets API. These offloaded functions are defined and registered, for example offload_socket() will be called when an application calls socket().

I did not find any sample applications for the modems, but as for example the SARA-R4 and the wncm14a2a do not expose any functions aside from socket offloading it can be assumed it is done in the same way.

Reply:
Aaah okay so these functions are the ones from include/net/socket.h ? 
Yeah that was exactly why I was asking. It seems tricky to expose such functions, but good to hear that there's another layer around them (if I understand correctly)